src/rocjpeg_vaapi_decoder.cpp

```diff
@@ -506,20 +500,7 @@ RocJpegStatus RocJpegVappiDecoder::CreateDecoderConfig() {
  */
 RocJpegStatus RocJpegVappiDecoder::CreateDecoderContext() {
 
-    uint32_t surface_format;
-    surface_format = VA_RT_FORMAT_YUV420;
-
-    VASurfaceAttrib surface_attrib;
-    surface_attrib.type = VASurfaceAttribPixelFormat;
-    surface_attrib.flags = VA_SURFACE_ATTRIB_SETTABLE;
-    surface_attrib.value.type = VAGenericValueTypeInteger;
-    surface_attrib.value.value.i = VA_FOURCC_NV12;
-
-    // Create a dummy surface with a resolution of min_picture_width_ x min_picture_height_ supported by the hardware.
-    // This surface is only used to create the context for the decoding pipeline, as context creation requires an initial surface.
-    // During the actual submission, the appropriate surfaces with the correct resolution will be created.
-    CHECK_VAAPI(vaCreateSurfaces(va_display_, surface_format, min_picture_width_, min_picture_height_, &va_surface_id_, 1, &surface_attrib, 1));
-    CHECK_VAAPI(vaCreateContext(va_display_, va_config_id_, min_picture_width_, min_picture_height_, VA_PROGRESSIVE, &va_surface_id_, 1, &va_context_id_));
+    CHECK_VAAPI(vaCreateContext(va_display_, va_config_id_, min_picture_width_, min_picture_height_, VA_PROGRESSIVE, nullptr, 0, &va_context_id_));
```

Comment:
Is Mesa ignoring the parameters `VASurfaceID *render_targets` and `int num_render_targets` during the `vaCreateContext` call a recent change? Previously, we encountered issues when we did not pass an initial surface for these two arguments. Could you please provide me with a link to the Mesa implementation of this function where these parameters are ignored?

Reply:
Mesa never used these parameters.

https://gitlab.freedesktop.org/mesa/mesa/-/blob/105fbaab4967efe22897faf0fe56af829a14273a/src/gallium/frontends/va/context.c#L281